R/save.r

```diff
@@ -58,6 +60,25 @@ ggsave <- function(filename, plot = last_plot(),
   invisible()
 }
 
+#' Parse a DPI input from the user
+#'
+#' Allows handling of special strings when user specifies a DPI like "print".
+#'
+#' @param dpi Input value from user
+#' @return Parsed DPI input value
+#' @noRd
+parse_dpi <- function(dpi) {
+  if (is.character(dpi)) {
+    dpi <- switch(dpi, screen = 72, print = 300, retina = 320, NULL)
+
+    if (is.null(dpi)) {
+      stop("Invalid string DPI value", call. = FALSE)
+    }
+  }
```

Comment:
You should have a branch here for `is.numeric()` that returns `dpi`, and then an `else` branch that throws an error saying DPI must be character or number.  You should also check that it's length 1.